jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/WriteTarFileStep.java

```diff
@@ -0,0 +1,104 @@
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+
+public class WriteTarFileStep implements AsyncStep<Void>, Callable<Void> {
+
+  private final Path outputPath;
+  private final BuildConfiguration buildConfiguration;
+  private final PullAndCacheBaseImageLayersStep pullAndCacheBaseImageLayersStep;
+  private final ImmutableList<BuildAndCacheApplicationLayerStep> buildAndCacheApplicationLayerSteps;
+  private final BuildImageStep buildImageStep;
+
+  private final ListeningExecutorService listeningExecutorService;
+  private final ListenableFuture<Void> listenableFuture;
+
+  WriteTarFileStep(
+      ListeningExecutorService listeningExecutorService,
+      Path outputPath,
+      BuildConfiguration buildConfiguration,
+      PullAndCacheBaseImageLayersStep pullAndCacheBaseImageLayersStep,
+      ImmutableList<BuildAndCacheApplicationLayerStep> buildAndCacheApplicationLayerSteps,
+      BuildImageStep buildImageStep) {
+    this.listeningExecutorService = listeningExecutorService;
+    this.buildConfiguration = buildConfiguration;
+    this.outputPath = outputPath;
+    this.pullAndCacheBaseImageLayersStep = pullAndCacheBaseImageLayersStep;
+    this.buildAndCacheApplicationLayerSteps = buildAndCacheApplicationLayerSteps;
+    this.buildImageStep = buildImageStep;
+
+    listenableFuture =
+        Futures.whenAllSucceed(
+                pullAndCacheBaseImageLayersStep.getFuture(), buildImageStep.getFuture())
+            .call(this, listeningExecutorService);
+  }
+
+  @Override
+  public ListenableFuture<Void> getFuture() {
+    return listenableFuture;
+  }
+
+  @Override
+  public Void call() throws ExecutionException, InterruptedException {
+    ImmutableList.Builder<ListenableFuture<?>> dependenciesBuilder = ImmutableList.builder();
+    for (PullAndCacheBaseImageLayerStep pullAndCacheBaseImageLayerStep :
+        NonBlockingSteps.get(pullAndCacheBaseImageLayersStep)) {
+      dependenciesBuilder.add(pullAndCacheBaseImageLayerStep.getFuture());
+    }
+    for (BuildAndCacheApplicationLayerStep buildAndCacheApplicationLayerStep :
+        buildAndCacheApplicationLayerSteps) {
+      dependenciesBuilder.add(buildAndCacheApplicationLayerStep.getFuture());
+    }
+    dependenciesBuilder.add(NonBlockingSteps.get(buildImageStep).getFuture());
+    return Futures.whenAllSucceed(dependenciesBuilder.build())
+        .call(this::afterPushBaseImageLayerFuturesFuture, listeningExecutorService)
+        .get();
+  }
+
+  private Void afterPushBaseImageLayerFuturesFuture() throws ExecutionException, IOException {
+    Image<CachedLayer> image = NonBlockingSteps.get(NonBlockingSteps.get(buildImageStep));
+
+    // Build the image to a tarball
+    buildConfiguration.getBuildLogger().lifecycle("Building image to tar file...");
+    Files.createDirectories(outputPath.getParent());
```

Comment:
In our use case, yes, the directory should already exist since we're putting it in the build output. But I imagine if this were eventually configurable, then it'd be a better experience to generate all the directories the user specifies rather than throwing an exception if the directory structure isn't already setup.